### DISCUSSION TITLE
Basic Help With Rules

### DISCUSSION THREAD

**Discussion by chrishome7** [arendst/Tasmota#14461] [Q&A] [answered]
Please forgive me, for I'm still learning.
Can Someone please help me compose a few rules for a Shelly 1.
I need the relay to disconnect if power or Wi-Fi is somehow lost and the relay cant be powered on for more then 30min straight.

Timer5 {"Enable":1,"Mode":0,___Some_stuff_maybe_____________"Repeat":probably,"Action": 3}
Rule1
On Wifi#Disconnected DO Power off ENDON ON 
On Mqtt#Disconnected DO Power off ENDON ON
ON clock#Timer=5 Do Power off ENDON ON

I'm confused so any guidance is deeply appreciated.

**Comment by barbudor** [comment, marked as answer]
> I need the relay to disconnect if power or Wi-Fi is somehow lost and the relay cant be powered on for more then 30min straight.

Quite a confusing sentence
"if ... Wifi is somehow lost" : ok
"if Power ... is somehow lost" ???? If you loose power to your device, Tasmota is off. Relay is likely to switch off too
"The relay can't be powered on for more than 30 min straight" : That one I don't get at all what you are meaning
If you can't power on the relay, it's already off isn't it ?

I believe you are not yet clean with exactly what you want.
One good way of thinking is trying to draw what you want as a state machine graph with:
- Stable states 
- Event
- Action on event and next stable state

**Reply by chrishome7** [reply]
Sorry I'm very new to all of this.

I simply want to compose rules so the shelly relay is turned off if WIFI if connections or Mqtt communications to the device are are lost.  IE, The device will still have power but the router goes out or my MQTT Broker is down.  If the relay is on and loses said communication wouldn't the relay not be left ON? If this is true then its is a very big problem for my application. 

State: relay is on
Event: WIFI and Or MQTT is Lost  
Action: Please turn the relay off
State Relay off

I also want the relay to only be actively ON for a period of no more then 30 min at a time. Currently HA handles this but  if communication is lost then I again have a problem. 

State: relay is on
Event: timer 30 min
Action: please turn the relay off
State Relay off

**Reply by barbudor** [reply]
> State: relay is on
Event: WIFI and Or MQTT is Lost
Action: Please turn the relay off
State Relay off

See Rules documention, events `on wifi#disconnected do ...`  and `on mqtt#disconnected do ... `

> State: relay is on
Event: timer 30 min
Action: please turn the relay off
State Relay off

No need for a timer
See command `PulseTime`

**Reply by chrishome7** [reply]
Thank you. For a laymen it can at times be slightly intimidating.
